tests/TestCase.php

```diff
@@ -33,17 +85,49 @@ final protected function assertExampleMatches(
     }
 
     $ast = parse($parser_ctx, $in);
-    $actual_html = (new HTMLRenderer($render_ctx))->render($ast);
+    foreach ($this->provideHTMLRendererConstructors() as list($constructor)) {
+      using (_Private\disable_child_validation()) {
+        // HHAST_FIXME[DontAwaitInALoop] Suboptimial, but it's test code...
+        $actual_html = await static::unsafeStringifyXHPChildAsync(
+          $constructor($render_ctx)->render($ast),
+        )
+          |> self::correctForSpecDeviations($$);
+      }
+      // Improve output readability
+      $actual_html = Str\replace($actual_html, "\t", self::TAB_REPLACEMENT);
+      $normalized_expected_html =
+        Str\replace($expected_html, "\t", self::TAB_REPLACEMENT);
 
-    // Improve output readability
-    $actual_html = Str\replace($actual_html, "\t", self::TAB_REPLACEMENT);
-    $expected_html = Str\replace($expected_html, "\t", self::TAB_REPLACEMENT);
+      expect($actual_html)->toBeSame(
+        $normalized_expected_html,
+        "HTML differs for %s:\n%s",
+        $name,
+        $in,
+      );
+    }
+  }
 
-    expect($actual_html)->toBeSame(
-      $expected_html,
-      "HTML differs for %s:\n%s",
-      $name,
-      $in,
-    );
+  private static function correctForSpecDeviations(string $html)[]: string {
+    return Str\replace_every($html, static::SLIGHT_DEVIATIONS_FROM_SPEC)
+      |> self::placeASolidusBeforeTheEndOfTheClosingImgTags($$);
+  }
+
+  private static function placeASolidusBeforeTheEndOfTheClosingImgTags(
+    string $html,
+  )[]: string {
+    // https://stackoverflow.com/posts/1732454/revisions
+    // #pragma enable module(guard(superstition))
+    return Regex\replace_with($html, re'#<img .+?>#', $he_comes ==> {
```

Comment:
If this joke is a little much, let me know and I'll rewrite the git history so it was never here to begin with.